mlrun/api/utils/builder.py

```diff
@@ -760,30 +762,36 @@ def resolve_image_target_and_registry_secret(
 
 def _generate_builder_env(
     project: str, builder_env: typing.Dict
-) -> typing.List[client.V1EnvVar]:
+) -> (typing.List[client.V1EnvVar], typing.List[client.V1EnvVar]):
     k8s = mlrun.api.utils.singletons.k8s.get_k8s_helper(silent=False)
     secret_name = k8s.get_project_secret_name(project)
     existing_secret_keys = k8s.get_project_secret_keys(project, filter_internal=True)
 
     # generate env list from builder env and project secrets
-    env = []
+    project_secrets = []
     for key in existing_secret_keys:
         if key not in builder_env:
             value_from = client.V1EnvVarSource(
                 secret_key_ref=client.V1SecretKeySelector(name=secret_name, key=key)
             )
-            env.append(client.V1EnvVar(name=key, value_from=value_from))
+            project_secrets.append(client.V1EnvVar(name=key, value_from=value_from))
+    env = []
     for key, value in builder_env.items():
         env.append(client.V1EnvVar(name=key, value=value))
-    return env
+    return env, project_secrets
 
 
-def _add_kaniko_args_with_all_build_args(args, builder_env, extra_args):
-
+def _add_kaniko_args_with_all_build_args(
+    args, builder_env, project_secrets, extra_args
+):
     builder_env = builder_env or []
+    project_secrets = project_secrets or []
 
     for env in builder_env:
-        args.extend(["--build-arg", f"{env.name}_ARG={env.value}"])
+        args.extend(["--build-arg", f"{env.name}={env.value}"])
+
+    for secret in project_secrets:
+        args.extend(["--build-arg", f"{secret.name}=${secret.name}"])
```

Comment:
```suggestion
        # using $ so that the value would not be a plain-text but read from the injected envvars
        args.extend(["--build-arg", f"{secret.name}=${secret.name}"])
```